jetbrains-core/tst/software/aws/toolkits/jetbrains/services/cloudwatch/logs/CloudWatchLogsServiceNodeTest.kt

```diff
@@ -0,0 +1,73 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cloudwatch.logs
+
+import com.intellij.testFramework.ProjectRule
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import software.amazon.awssdk.services.cloudwatchlogs.model.LogGroup
+import software.aws.toolkits.jetbrains.core.MockResourceCache
+import software.aws.toolkits.jetbrains.core.explorer.nodes.AwsExplorerEmptyNode
+import software.aws.toolkits.jetbrains.core.explorer.nodes.AwsExplorerErrorNode
+import software.aws.toolkits.jetbrains.core.explorer.nodes.CloudWatchRootNode
+import software.aws.toolkits.jetbrains.services.cloudwatch.logs.resources.CloudWatchResources
+import java.util.concurrent.CompletableFuture
+
+class CloudWatchLogsServiceNodeTest {
+    @JvmField
+    @Rule
+    val projectRule = ProjectRule()
+
+    @Before
+    fun setUp() {
+        resourceCache().clear()
+    }
+
+    @Test
+    fun logGroupsAreListed() {
+        resourceCache().logGroups(listOf("bcd", "abc", "zzz", "AEF"))
+
+        val children = CloudWatchLogsServiceNode(projectRule.project, CLOUDWATCH_LOGS_EXPLORER_SERVICE_NODE).children
+
+        assertThat(children).allMatch { it is CloudWatchLogsNode }
+        assertThat(children.filterIsInstance<CloudWatchLogsNode>().map { it.displayName() }).containsExactlyInAnyOrder("abc", "AEF", "bcd", "zzz")
```

Comment:
we should probably have a deterministic sort order across all user resources (incl. credential profiles)

Reply:
Well it's funny you mention that because we do (through the very verbose `DefaultAwsExplorerTreeStructureProvider`) will change test